### PR TITLE
Jk/feature/postcard preview

### LINKED ISSUE #38: EXTENSION Postcard preview back button
- [ ] Retain postcard message/image data, so users can click back button and edit what they've already added

### PR DESCRIPTION
**Pull request by JeffKersting**
#### Purpose:
 > Check all that apply.
- [x] Feature
- [ ] Refactor
- [ ] Bug Fix 
- [ ] Clean Up

### What does this PR do?
 > Clearly and concisely describe changes made. 

Adds styling to the postcard preview form. Adds an edit button that passes the users postcard data back to the form page. Additionally adds minor styling to the postcard form page.

Issues closed #38 #21 #20

N/A

> Describe your code changes in detail for reviewers.

Update preview-page sass file to add styling. Add back button to preview page.

Postcard-Preview.js. _Postcard-Preview.scss. Postcard-From.js. _postcard-form.scss.

In browser, ensure that site navigation is still functional, and styling is applied properly.

Implement graphQL.

### CODE REVIEW

File: src/components/Postcard-Form/Postcard-Form.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+          <button type='button' id='clear-image' onClick={() => setImage('')}>X</button>
+        </div>
+        <section className='image-thumbnail'>
+          {image && <img src={image} alt='image thumbnail' />}
+        </section>
+        <h3>Enter your message below</h3>
+        <label htmlFor='message' value='message' />
+        <textarea type='text' maxLength='400' id='message' value={message} required/>
+        <label htmlFor='preview' value='preview postcard' />
+        <Link to={{
```
Comment (gaj23) [inline src/components/Postcard-Form/Postcard-Form.js:49]: I really enjoy the use of Router like this. I think it's a powerful way to blend technologies.

File: src/components/Postcard-Preview/Postcard-Preview.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        <Link to={{
+          pathname: '/create-postcard',
+          state: {
+            image,
+            message
+          }
+        }}>
+        <button>Edit</button>
+        </Link>
+        <button>Send!</button>
```
Comment (gaj23) [inline src/components/Postcard-Preview/Postcard-Preview.js:43]: We'll be able to add a success message to this and then after a time out, redirect back to create a postcard sounds good to me.

File: src/components/Postcard-Preview/_Postcard-Preview.scss [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+  .front {
+    border: 1px solid $ink-blue;
+    margin: 1rem;
+    padding: .5rem .35rem .5rem;
+    width: 95%;
+    height: 40vh;
+    object-fit: cover;
+    overflow: hidden;
+    background: $off-white;
```
Comment (gaj23) [inline src/components/Postcard-Preview/_Postcard-Preview.scss:17]: Was this color used to differentiate between our "white" and the postcard itself?

File: src/components/User-Form/Update-Form.js [hunk 4]
```diff
@@ -58,8 +59,22 @@ const UpdateUserInfoForm = ({address, userAbout, updateHandler, back}) => {
...
         <label htmlFor='back' value='back' />
         <button onClick={back} id='back'>Back</button>
-    </div>
+    </section>
   )
 }
 
-export default UpdateUserInfoForm
+UpdateUserInfoForm.propTypes = {
+  address: PropTypes.shape({
```
Comment (gaj23) [inline src/components/User-Form/Update-Form.js:67]: I didn't know about the 'shape' property of propTypes. This was great to learn.

File: src/components/User-Landing-Page/User-Landing.js [hunk 5]
```diff
@@ -53,7 +54,9 @@ const UserLandingPage = () => {
                 <h4 className='connection-name'>You are connected with {connection.name} ({connection.country})</h4>
                 <h4 className='about'>{connection.about}</h4>
                 <div className='button-wrapper'>
-                  <button>Send Postcard</button>
+                  <Link to='/create-postcard'>
+                    <button>Send Postcard</button>
+                  </Link>
                   <button>End Connection</button>
```
Comment (gaj23) [inline src/components/User-Landing-Page/User-Landing.js:60]: Maybe we should change the wording for end connection to something like "Go on Hiatus" or "Get a new Buddy" or something?

**Review by gaj23 — APPROVED** [review]
This is a great compilation of our work that we've done together. Very nice to have the wire frame complete. Thank you for finishing the last portion of it, Jeff.

File: src/components/Postcard-Form/Postcard-Form.js (hunk 1 above)
Comment (kelsiebesingeryeh) [reply src/components/Postcard-Form/Postcard-Form.js:49]: same

File: src/components/Postcard-Form/_postcard-form.scss [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    width: 25vw;
+    margin-top: 2em;
+
+    img {
+      height: 100%;
+      width: 100%;
+    }
+  }
+
+  #preview {
```
Comment (kelsiebesingeryeh) [inline src/components/Postcard-Form/_postcard-form.scss:34]: don't think we're supposed to use IDs for styling, should be a classs

Comment (gaj23) [reply src/components/Postcard-Form/_postcard-form.scss:34]: Good point Kelsie!

**Review by kelsiebesingeryeh — APPROVED** [review]

**Pull request merged**